### PR TITLE
fix: back-compat issues with cloud-profiler

### LINKED ISSUE #123: Problems migrating to v0.18.X
Right now, we're using ServiceObject.request() with a callback so we can get the error, body, and response body even when there's an error. We need to do this because the profiler API will return an error code, with a backoff specified in the body.

What can we do to migrate?

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Resolves #123

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
@nolanmar511 this should fix what you're running into :)

File: src/service-object.ts [hunk 1]
```diff
@@ -430,7 +430,16 @@ class ServiceObject extends EventEmitter {
    * @param {string} reqOpts.uri - A URI relative to the baseUrl.
    * @param {function} callback - The callback function passed to `request`.
    */
-  request(reqOpts: DecorateRequestOptions): Promise<r.Response> {
+  request(reqOpts: DecorateRequestOptions): Promise<r.Response>;
+  request(reqOpts: DecorateRequestOptions, callback: r.RequestCallback): void;
+  request(reqOpts: DecorateRequestOptions, callback?: r.RequestCallback):
+      void|Promise<r.Response> {
+    if (callback) {
+      this.request_(reqOpts).then(
```
Comment (ofrobots) [inline src/service-object.ts]: infinitesimally teeny nit. If you reverse the direction of this branch, you will need one less `return`.

File: src/service-object.ts [hunk 2]
```diff
@@ -430,7 +430,16 @@ class ServiceObject extends EventEmitter {
...
+          r => callback(null, r as r.Response, r.body),
```
Comment (ofrobots) [inline src/service-object.ts]: This line has a variable named `r` and a type named `r`. Can you remove the ambiguity even if the compiler / linter isn't complaining about it.

**Comment by JustinBeckwith** [comment]
Added a test 🎉

**Pull request merged**